<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>qseditoptions</class>
 <widget class="QDialog" name="qseditoptions">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>638</width>
    <height>672</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Preferences</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QTabWidget" name="tabWidget">
     <property name="toolTip">
      <string>Provides a table of MIDI output ports/busses and clock 
settings, input busses, and more.  There is a bug in the
MIDI Clock tab where it won't respond to a click except
in a very specific place; the alternative is to do Alt-C
once or twice. </string>
     </property>
     <property name="currentIndex">
      <number>2</number>
     </property>
     <widget class="QWidget" name="tabMidiClock">
      <property name="toolTip">
       <string>Provides a list of MIDI output ports and clock settings.</string>
      </property>
      <attribute name="title">
       <string>MIDI &amp;Clock</string>
      </attribute>
      <widget class="QGroupBox" name="groupBoxMetaEvents">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>400</y>
         <width>591</width>
         <height>73</height>
        </rect>
       </property>
       <property name="title">
        <string>Meta Events</string>
       </property>
       <widget class="QPlainTextEdit" name="plainTextEditTempoTrack">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>30</y>
          <width>42</width>
          <height>28</height>
         </rect>
        </property>
        <property name="toolTip">
         <string>Sets the number of the tempo track. Button not yet functional.</string>
        </property>
        <property name="verticalScrollBarPolicy">
         <enum>Qt::ScrollBarAlwaysOff</enum>
        </property>
        <property name="horizontalScrollBarPolicy">
         <enum>Qt::ScrollBarAlwaysOff</enum>
        </property>
        <property name="readOnly">
         <bool>false</bool>
        </property>
        <property name="overwriteMode">
         <bool>false</bool>
        </property>
        <property name="placeholderText">
         <string>0</string>
        </property>
       </widget>
       <widget class="QLabel" name="textLabelTempoTrack">
        <property name="geometry">
         <rect>
          <x>64</x>
          <y>32</y>
          <width>341</width>
          <height>21</height>
         </rect>
        </property>
        <property name="toolTip">
         <string>Not yet functional.</string>
        </property>
        <property name="text">
         <string>Tempo track number (0 to 1023, 0 recommended)</string>
        </property>
       </widget>
       <widget class="QPushButton" name="pushButtonTempoTrack">
        <property name="geometry">
         <rect>
          <x>420</x>
          <y>30</y>
          <width>162</width>
          <height>32</height>
         </rect>
        </property>
        <property name="toolTip">
         <string>Saves the current tempo track number to the MIDI file. 
Button not yet functional.</string>
        </property>
        <property name="text">
         <string>Set as Tempo Track</string>
        </property>
       </widget>
      </widget>
      <widget class="QWidget" name="layoutWidgetClockStart">
       <property name="geometry">
        <rect>
         <x>30</x>
         <y>352</y>
         <width>338</width>
         <height>30</height>
        </rect>
       </property>
       <layout class="QHBoxLayout" name="horizLayoutClockStartModulo">
        <item>
         <widget class="QLabel" name="labelClockStartModulo">
          <property name="toolTip">
           <string>Sets the position after which MIDI clocking begins.</string>
          </property>
          <property name="text">
           <string>Clock Start Modulo (1/16 Notes)</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSpinBox" name="spinBoxClockStartModulo">
          <property name="minimum">
           <number>1</number>
          </property>
          <property name="maximum">
           <number>1024</number>
          </property>
          <property name="value">
           <number>64</number>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QPushButton" name="pushButtonStoreMap">
       <property name="geometry">
        <rect>
         <x>432</x>
         <y>352</y>
         <width>162</width>
         <height>32</height>
        </rect>
       </property>
       <property name="toolTip">
        <string>Saves maps of I/O port numbers and names.  At exit,
these maps are saved to the 'rc' file. Then, each port
number can be found by name-lookup, in cases where
the system's MIDI setup has changed.</string>
       </property>
       <property name="text">
        <string>Save Clock/Input Maps</string>
       </property>
      </widget>
      <widget class="QLabel" name="label">
       <property name="geometry">
        <rect>
         <x>72</x>
         <y>490</y>
         <width>161</width>
         <height>17</height>
        </rect>
       </property>
       <property name="text">
        <string>Seq66's MIDI Client ID</string>
       </property>
      </widget>
      <widget class="QPlainTextEdit" name="plainTextEditClientId">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>480</y>
         <width>42</width>
         <height>28</height>
        </rect>
       </property>
       <property name="toolTip">
        <string>Holds the ALSA MIDI client number, if applicable.</string>
       </property>
       <property name="verticalScrollBarPolicy">
        <enum>Qt::ScrollBarAlwaysOff</enum>
       </property>
       <property name="horizontalScrollBarPolicy">
        <enum>Qt::ScrollBarAlwaysOff</enum>
       </property>
       <property name="readOnly">
        <bool>true</bool>
       </property>
       <property name="placeholderText">
        <string>0</string>
       </property>
      </widget>
      <widget class="QCheckBox" name="outPortsMappedCheck">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="geometry">
        <rect>
         <x>432</x>
         <y>488</y>
         <width>177</width>
         <height>23</height>
        </rect>
       </property>
       <property name="toolTip">
        <string>If checked, an output port map was loaded.</string>
       </property>
       <property name="text">
        <string>Output Ports Mapped</string>
       </property>
      </widget>
      <widget class="QLabel" name="portsClocksLabel">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>10</y>
         <width>131</width>
         <height>17</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <family>DejaVu Sans</family>
         <weight>75</weight>
         <bold>true</bold>
        </font>
       </property>
       <property name="text">
        <string>Ports and Clocks</string>
       </property>
      </widget>
      <widget class="QScrollArea" name="clocksScrollArea">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>40</y>
         <width>591</width>
         <height>301</height>
        </rect>
       </property>
       <property name="frameShape">
        <enum>QFrame::NoFrame</enum>
       </property>
       <property name="frameShadow">
        <enum>QFrame::Plain</enum>
       </property>
       <property name="horizontalScrollBarPolicy">
        <enum>Qt::ScrollBarAlwaysOff</enum>
       </property>
       <property name="widgetResizable">
        <bool>true</bool>
       </property>
       <widget class="QWidget" name="scrollAreaWidgetContents">
        <property name="geometry">
         <rect>
          <x>0</x>
          <y>0</y>
          <width>591</width>
          <height>301</height>
         </rect>
        </property>
       </widget>
      </widget>
     </widget>
     <widget class="QWidget" name="tabMidiInput">
      <property name="toolTip">
       <string>Provides a table to select the desired MIDI input port.</string>
      </property>
      <attribute name="title">
       <string>MIDI &amp;Input</string>
      </attribute>
      <widget class="QGroupBox" name="groupBoxInputOptions">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>420</y>
         <width>591</width>
         <height>81</height>
        </rect>
       </property>
       <property name="title">
        <string>Input Options</string>
       </property>
       <widget class="QCheckBox" name="checkBoxRecordByChannel">
        <property name="geometry">
         <rect>
          <x>20</x>
          <y>40</y>
          <width>561</width>
          <height>21</height>
         </rect>
        </property>
        <property name="toolTip">
         <string>If checked, incoming data is routed to the desired pattern
number using the channel number (0 to 15).</string>
        </property>
        <property name="text">
         <string>Record input into sequences according to channel</string>
        </property>
       </widget>
      </widget>
      <widget class="QCheckBox" name="inPortsMappedCheck">
       <property name="enabled">
        <bool>false</bool>
       </property>
       <property name="geometry">
        <rect>
         <x>424</x>
         <y>328</y>
         <width>169</width>
         <height>23</height>
        </rect>
       </property>
       <property name="text">
        <string>Input Ports Mapped</string>
       </property>
      </widget>
      <widget class="QLabel" name="inputBussesLabel">
       <property name="geometry">
        <rect>
         <x>16</x>
         <y>8</y>
         <width>97</width>
         <height>17</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <family>DejaVu Sans</family>
         <weight>75</weight>
         <bold>true</bold>
        </font>
       </property>
       <property name="text">
        <string>Input Buses</string>
       </property>
      </widget>
      <widget class="QScrollArea" name="inputsScrollArea">
       <property name="geometry">
        <rect>
         <x>16</x>
         <y>32</y>
         <width>585</width>
         <height>281</height>
        </rect>
       </property>
       <property name="frameShape">
        <enum>QFrame::NoFrame</enum>
       </property>
       <property name="frameShadow">
        <enum>QFrame::Plain</enum>
       </property>
       <property name="horizontalScrollBarPolicy">
        <enum>Qt::ScrollBarAlwaysOff</enum>
       </property>
       <property name="widgetResizable">
        <bool>true</bool>
       </property>
       <widget class="QWidget" name="scrollAreaWidgetContents_2">
        <property name="geometry">
         <rect>
          <x>0</x>
          <y>0</y>
          <width>585</width>
          <height>281</height>
         </rect>
        </property>
       </widget>
      </widget>
     </widget>
     <widget class="QWidget" name="tabDisplay">
      <property name="toolTip">
       <string>Provides some minor tweaks of the user-interface.  Edit the 'usr' file for far more control over settings.</string>
      </property>
      <attribute name="title">
       <string>D&amp;isplay</string>
      </attribute>
      <widget class="QGroupBox" name="groupBoxBooleanOptions">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>128</y>
         <width>361</width>
         <height>80</height>
        </rect>
       </property>
       <property name="title">
        <string>UI Boolean Options</string>
       </property>
       <widget class="QCheckBox" name="checkBoxPaletteActive">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>40</y>
          <width>261</width>
          <height>21</height>
         </rect>
        </property>
        <property name="toolTip">
         <string>If checked, the given palette file will be read at startup.
This option is saved in the 'rc' file.</string>
        </property>
        <property name="text">
         <string>Use Palette File</string>
        </property>
       </widget>
      </widget>
      <widget class="QWidget" name="layoutWidgetGuiOptions">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>10</y>
         <width>313</width>
         <height>64</height>
        </rect>
       </property>
       <layout class="QGridLayout" name="gridLayoutUiValues">
        <item row="1" column="0">
         <widget class="QLabel" name="labelUiScaling">
          <property name="text">
           <string>Scaling of User Interface *</string>
          </property>
         </widget>
        </item>
        <item row="1" column="2">
         <widget class="QLabel" name="label_2">
          <property name="text">
           <string>X</string>
          </property>
         </widget>
        </item>
        <item row="0" column="0">
         <widget class="QLabel" name="labelEditorKeyHeight">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Minimum">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="toolTip">
           <string>Sets the height of a note key in pixels.  
A primitive way to zoom.</string>
          </property>
          <property name="text">
           <string>Editor Key Height</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QSpinBox" name="spinKeyHeight">
          <property name="minimum">
           <number>4</number>
          </property>
          <property name="maximum">
           <number>24</number>
          </property>
          <property name="value">
           <number>10</number>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QLineEdit" name="lineEditUiScaling">
          <property name="toolTip">
           <string>Scales the main window at startup.  This is the width
scale factor, ranging from 0.5 to 3.0.

Use 0.85 at a minimum.</string>
          </property>
          <property name="text">
           <string>1.0</string>
          </property>
         </widget>
        </item>
        <item row="1" column="3">
         <widget class="QLineEdit" name="lineEditUiScalingHeight">
          <property name="toolTip">
           <string>Scales the main window at startup.  This is the height
scale factor, ranging from 0.5 to 3.0.

Use 0.60 at a minimum.</string>
          </property>
          <property name="text">
           <string>1.0</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QLabel" name="labelRestartNeeded">
       <property name="geometry">
        <rect>
         <x>416</x>
         <y>0</y>
         <width>193</width>
         <height>16</height>
        </rect>
       </property>
       <property name="text">
        <string>* Restart of Seq66 required</string>
       </property>
      </widget>
      <widget class="QWidget" name="layoutWidgetOp">
       <property name="geometry">
        <rect>
         <x>8</x>
         <y>88</y>
         <width>505</width>
         <height>29</height>
        </rect>
       </property>
       <layout class="QHBoxLayout" name="horizontalLayout">
        <item>
         <widget class="QLabel" name="labelPaletteFile">
          <property name="text">
           <string>Palette File Base Name</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="lineEditPaletteFile">
          <property name="toolTip">
           <string>Provides the base name of the palette file.  This file
always resides in the Seq66 configuration directory
(~/.config/seq66 or the NSM session directory). Any
path added to this name is stripped.</string>
          </property>
          <property name="text">
           <string>qseq66.palette</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="pushButtonSavePalette">
          <property name="text">
           <string>&amp;Save Current Palette</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
     <widget class="QWidget" name="tabJackSync">
      <property name="toolTip">
       <string>Provides settings for JACK transport and JACK MIDI.</string>
      </property>
      <attribute name="title">
       <string>&amp;JACK</string>
      </attribute>
      <widget class="QGroupBox" name="r_group_jack_transport">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>10</y>
         <width>591</width>
         <height>151</height>
        </rect>
       </property>
       <property name="title">
        <string>JACK Transport/MIDI</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop</set>
       </property>
      </widget>
      <widget class="QPushButton" name="btnJackConnect">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>290</y>
         <width>281</width>
         <height>23</height>
        </rect>
       </property>
       <property name="toolTip">
        <string>Connect/reconnect to JACK transport.</string>
       </property>
       <property name="text">
        <string>JACK Transport Co&amp;nnect</string>
       </property>
      </widget>
      <widget class="QPushButton" name="btnJackDisconnect">
       <property name="geometry">
        <rect>
         <x>310</x>
         <y>290</y>
         <width>291</width>
         <height>23</height>
        </rect>
       </property>
       <property name="toolTip">
        <string>Disconnect from JACK transport.</string>
       </property>
       <property name="text">
        <string>JACK Transport &amp;Disconnect</string>
       </property>
      </widget>
      <widget class="QWidget" name="layoutWidgetJackTransport">
       <property name="geometry">
        <rect>
         <x>30</x>
         <y>40</y>
         <width>319</width>
         <height>120</height>
        </rect>
       </property>
       <layout class="QVBoxLayout" name="vertLayoutJackTransportMidi">
        <item>
         <widget class="QCheckBox" name="chkJackConditional">
          <property name="toolTip">
           <string>Sets Seq66 as the MIDI Master if no master exists.</string>
          </property>
          <property name="text">
           <string>Master Conditional</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QCheckBox" name="chkJackMaster">
          <property name="toolTip">
           <string>Sets Seq66 as the MIDI Master.</string>
          </property>
          <property name="text">
           <string extracomment="Serve as jack transport master">Transport Master</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QCheckBox" name="chkJackTransport">
          <property name="toolTip">
           <string>Turns on Seq66 JACK transport as a client.</string>
          </property>
          <property name="text">
           <string>Enable Jack Transport</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QCheckBox" name="chkJackNative">
          <property name="toolTip">
           <string>Requests Seq66 to use the JACK system if it is running.
Otherwise, ALSA is used.  Not available on Windows/Mac.</string>
          </property>
          <property name="text">
           <string>Native JACK &amp;MIDI (requires a restart)</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QGroupBox" name="r_group_jack_start">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>180</y>
         <width>591</width>
         <height>91</height>
        </rect>
       </property>
       <property name="title">
        <string>JACK Start</string>
       </property>
       <property name="checkable">
        <bool>false</bool>
       </property>
      </widget>
      <widget class="QWidget" name="layoutWidgetJack">
       <property name="geometry">
        <rect>
         <x>31</x>
         <y>210</y>
         <width>104</width>
         <height>54</height>
        </rect>
       </property>
       <layout class="QVBoxLayout" name="radio_group_layout">
        <item>
         <widget class="QRadioButton" name="radio_live_mode">
          <property name="toolTip">
           <string>Play the MIDI song in Live Mode, where each pattern
plays in an endless loop.</string>
          </property>
          <property name="text">
           <string>&amp;Live Mode</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QRadioButton" name="radio_song_mode">
          <property name="toolTip">
           <string>Play the MIDI song in Song Mode, where the playback of
patterns is controlled by the layout drawn in the Song Editor.</string>
          </property>
          <property name="text">
           <string>&amp;Song Mode</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
     <widget class="QWidget" name="tabPlayOptions">
      <property name="toolTip">
       <string>Provides play options.  Minimal at present.</string>
      </property>
      <attribute name="title">
       <string>&amp;Play Options</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_3">
       <item>
        <widget class="QGroupBox" name="groupBox">
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>100</height>
          </size>
         </property>
         <property name="title">
          <string>Flags</string>
         </property>
         <widget class="QCheckBox" name="chkNoteResume">
          <property name="geometry">
           <rect>
            <x>24</x>
            <y>32</y>
            <width>355</width>
            <height>23</height>
           </rect>
          </property>
          <property name="toolTip">
           <string>When checked, unfinished notes are turned back on when
playback resumes.</string>
          </property>
          <property name="text">
           <string>&amp;Resume Note Ons at start/stop or sequence toggle</string>
          </property>
         </widget>
         <widget class="QCheckBox" name="chkUseFilesPPQN">
          <property name="geometry">
           <rect>
            <x>24</x>
            <y>56</y>
            <width>353</width>
            <height>23</height>
           </rect>
          </property>
          <property name="toolTip">
           <string>When checked, the PPQN of the MIDI file being loaded
overrides the default Seq66 PPQN.  Otherwise, the
file's PPQN is scaled to the default PPQN.</string>
          </property>
          <property name="text">
           <string>Use File's PPQN for Pre-Existing Files</string>
          </property>
         </widget>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_ppqn">
         <property name="maximumSize">
          <size>
           <width>120</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="font">
          <font>
           <family>DejaVu Sans</family>
           <weight>75</weight>
           <bold>true</bold>
          </font>
         </property>
         <property name="text">
          <string>PPQN Settings</string>
         </property>
        </widget>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_3">
         <item>
          <widget class="QComboBox" name="combo_box_ppqn">
           <property name="maximumSize">
            <size>
             <width>100</width>
             <height>16777215</height>
            </size>
           </property>
           <property name="toolTip">
            <string>Changes the default PPQN from 192 to another value.
This new default value is saved in the 'usr' file when
Seq66 exits.</string>
           </property>
           <property name="editable">
            <bool>true</bool>
           </property>
           <property name="currentText">
            <string>9600</string>
           </property>
           <property name="maxCount">
            <number>32</number>
           </property>
           <property name="insertPolicy">
            <enum>QComboBox::NoInsert</enum>
           </property>
           <property name="frame">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="label_ppqn_2">
           <property name="minimumSize">
            <size>
             <width>460</width>
             <height>0</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>200</width>
             <height>16777215</height>
            </size>
           </property>
           <property name="text">
            <string>Default Seq66 PPQN for New or Converted Files</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <spacer name="verticalSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeType">
          <enum>QSizePolicy::MinimumExpanding</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="buttonBoxOptionsDialog">
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>
